### DISCUSSION TITLE
Wrapper to fit regressions

### DISCUSSION THREAD

**Discussion by DominiqueMakowski** [easystats/easystats#91] [Ideas]
https://github.com/easystats/parameters/discussions/361#discussioncomment-190236 made me think, whether it would make sense to add a little "wrapper" for regressions...

Basically it would just use a decision algorithm based on the formula, like:

- if basic, use lm(),
- if family != gaussian, use glm()
- if presence of random factors, use glmmTMB
- if smooth terms, use gamm4 / mgcv
- if `bayesian=T`, use the according rstanarm functions

It could be quite basic and work for the most basic usecases, as this would be a convenience wrapper rather than a cutting-edge expert-level feature. Benefits could be native full easystats support, like we could clean/tidy up the output so that it's consistent no-matter what package was internally used. 

I mean, at least I would use it since for 99% of my work I use the models above 😁 so it would visually please me to always use the same function to fit models (+ in a pipe friendly way)

**Comment by mattansb** [comment]
I bet you can guess what I think of this even without me saying 😅

**Reply by DominiqueMakowski** [reply]
I'm sure you find it absolutely fantastic to facilitate users' life and let them mindlessly run stuff without any idea of what's happening behind the curtains 👍

**Reply by mattansb** [reply]
😂😂😂😂

**Comment by strengejacke** [comment]
Yeah, sounds good. We just need to clarify how to implement the formula-feature:

1. lme-style (with separated random effects notation) or lme4-style? thus, for mixed models, we need to re-arrange to formula to bring it in shape
2. all those economists FE regression, where the formula may have two parts (fixed and instruments), or three parts (fixed, instruments and clusters). 
3. How to decide which kind of model the user wants to fit when we have: `art ~ fem + mar + kid5 + phd + ment | fem + mar + kid5 + phd + ment` - is it a FE model, or a zero-inflated model?
4. How to deal with zero-inflation formula for mixed effects
5. As for glm, which family and which link-function to use?
6. how to include dispersion-formulas?
7 ....

This list is probably much longer, but you may try to write a first draft (which might end up in rewriting the easystats packages 😬 ).

![homer_dimishing](https://user-images.githubusercontent.com/26301769/101878960-ca485700-3b90-11eb-9d0e-bf6810bc6ae0.gif)

**Comment by DominiqueMakowski** [comment]
for the formula, I'd go the brms/lme4 way and having every part in the formula, I personally hate when random effects are a separate argument (for the zi and what not, no idea, as I never use them so I trust your judgment)

> all those economists FE regression, where the formula may have two parts

I'm a simple psychologist i don't ~want to~ know what you're talking about 😁 

> As for glm, which family and which link-function to use?

I think by default the family is gaussian-identity, and if different the family arg is simply passed down to the relevant function (wether glm() or glmer() etc)

**Reply by strengejacke** [reply]
> I'd go the brms/lme4 way and having every part in the formula

Then you have to rewrite/reconstruct the formula when you want to use `lme()`, `gls()`, `mgcv()`... In either way, I don't see an easy way here, because to be sure which kind of model to fit, we cannot rely on the formula. We have to add many more arguments, which ends up in being more complicated then simply fitting the model in a "natural" way. The idea you described works best for a limited set of models.

It sounds to me like a similar approach to [recipes](https://recipes.tidymodels.org/), however, I personally am no fan of this kind of modelling "syntax".
